### PR TITLE
Another NERF to heavy revolver ammo

### PR DESCRIPTION
**Pull request by ihatethisengine**
# About the pull request
https://github.com/cmss13-devs/cmss13/pull/4706
So this PR replaced knockdown with slow+knockback, but unlike the knockdown proc the slow proc didn't check for the mob size, so you became able to slow down T3s and even Queen with heavy revolver ammo, which clearly wasn't intended. This wasn't noticed until recently.

# Explain why it's good for the game
Free slows with no telegraph on T3s is not something we need in the game.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: ihatethisengine2
balance: heavy revolver ammo cannot slowdown t3s anymore.
/:cl:

### CODE REVIEW

File: code/datums/ammo/ammo.dm [hunk 1]
```diff
@@ -173,6 +173,8 @@
 		living_mob.apply_stamina_damage(fired_projectile.ammo.damage, fired_projectile.def_zone, ARMOR_BULLET)
 
 /datum/ammo/proc/slowdown(mob/living/living_mob, obj/projectile/fired_projectile)
+	if(living_mob.mob_size >= MOB_SIZE_BIG)
+        return //Big mobs are not affected.
```
Comment (ihatethisengine) [inline code/datums/ammo/ammo.dm:177]:
```suggestion
	if(living_mob.mob_size >= MOB_SIZE_BIG)
		return //Big mobs are not affected.
```

**Comment by joookschad** [comment]
this removes the only good part of heavy revolver ammo

**Comment by ihatethisengine** [comment]
> this removes the only good part of heavy revolver ammo

I understand that you are probably joking, but chainslowing T3s with IFF and ZERO telegraph cannot be justified.

File: code/datums/ammo/ammo.dm [hunk 2]
```diff
@@ -173,6 +173,8 @@
 		living_mob.apply_stamina_damage(fired_projectile.ammo.damage, fired_projectile.def_zone, ARMOR_BULLET)
 
 /datum/ammo/proc/slowdown(mob/living/living_mob, obj/projectile/fired_projectile)
+	if(living_mob.mob_size >= MOB_SIZE_BIG)
```
Comment (Drulikar) [inline code/datums/ammo/ammo.dm:176]: some t2s are `MOB_SIZE_BIG`
consider instead type checking to ensure its a xeno and then check `tier`.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/datums/ammo/ammo.dm [hunk 3]
```diff
@@ -173,6 +173,8 @@
...
+		return //Big mobs are not affected.
```
Comment (ihatethisengine) [inline code/datums/ammo/ammo.dm:177]:
```suggestion
```

File: code/datums/ammo/ammo.dm (hunk 2 above)
Comment (ihatethisengine) [inline code/datums/ammo/ammo.dm:176]:
```suggestion
	if(isxeno(living_mob))
		var/mob/living/carbon/xenomorph/xeno = living_mob
		if(xeno.caste.tier > 2)
				return //tier 3 are not affected
```

Comment (ihatethisengine) [reply code/datums/ammo/ammo.dm]: Sure but slugs for example don't affect carriers, hivelords etc.

File: code/datums/ammo/ammo.dm [hunk 4]
```diff
@@ -173,6 +173,10 @@
 		living_mob.apply_stamina_damage(fired_projectile.ammo.damage, fired_projectile.def_zone, ARMOR_BULLET)
 
 /datum/ammo/proc/slowdown(mob/living/living_mob, obj/projectile/fired_projectile)
+	if(isxeno(living_mob))
+		var/mob/living/carbon/xenomorph/xeno = living_mob
+		if(xeno.caste.tier > 2)
+				return //tier 3 are not affected
```
Comment (ihatethisengine) [inline code/datums/ammo/ammo.dm:179]:
```suggestion
			return //tier 3 are not affected
```

File: code/datums/ammo/ammo.dm [hunk 5]
```diff
@@ -173,6 +173,10 @@
 		living_mob.apply_stamina_damage(fired_projectile.ammo.damage, fired_projectile.def_zone, ARMOR_BULLET)
 
 /datum/ammo/proc/slowdown(mob/living/living_mob, obj/projectile/fired_projectile)
+	if(isxeno(living_mob))
+		var/mob/living/carbon/xenomorph/xeno = living_mob
+		if(xeno.caste.tier > 2)
+			return //tier 3 are not affected
```
Comment (Drulikar) [inline code/datums/ammo/ammo.dm:179]: This should work but you likely don't want queen to be affected either so needs additional checks.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/datums/ammo/ammo.dm (hunk 5 above)
Comment (Drulikar) [reply code/datums/ammo/ammo.dm:179]: Feel free to tweak slugs too if you don't like them behaving different.

Comment (ihatethisengine) [inline code/datums/ammo/ammo.dm:179]:
```suggestion
		if(xeno.caste.tier > 2 || (xeno.caste.tier == 0 && xeno.mob_size >= MOB_SIZE_BIG))
			return //tier 3 and big tier 0 (like queen) are not affected
```

**Review by Drulikar — APPROVED** [review]

**Comment by Drulikar** [comment]
Feel free to tweak slugs for consistency. Also I will note that predaliens are tier 1 so will be slowed still.

**Pull request merged**